src/main/java/seedu/address/model/location/Location.java

```diff
@@ -0,0 +1,148 @@
+    private static final double MIN_LON = 103.61751;
+    private static final double MAX_LON = 104.04360;
+    private static final double ALLOWABLE_ERROR = 0.0001;
+
+    private static final String DOUBLE_PATTERN = "[0-9]+(\\.)?[0-9]*";
+
+    private final String name;
+    private final double lat;
+    private final double lon;
+
+    /**
+     * Constructs an unnamed {@code Location}.
+     * @param lat A valid latitude.
+     * @param lon A valid longitude.
+     */
+    public Location(double lat, double lon) {
+        this("", lat, lon);
+    }
+
+    /**
+     * Constructs a named {@code Location}.
+     * @param name A non-null string.
+     * @param lat A valid latitude.
+     * @param lon A valid longitude.
+     */
+    public Location(String name, double lat, double lon) {
+        requireAllNonNull(name, lat, lon);
+        checkArgument(isValidLocation(lat, lon), MESSAGE_CONSTRAINTS);
+        this.name = name;
+        this.lat = lat;
+        this.lon = lon;
+    }
+
+    /**
+     * Checks whether the location is valid.
+     * We also check whether the lat and lon can be parsed into a {@code Double}
+     * so that we don't have to error-handle later on.
+     * @param lat A string representation of the latitude.
+     * @param lon A string representation of the longitude.
+     * @return whether the lat and lon fit the requirements.
+     */
+    public static boolean isValidLocation(String lat, String lon) {
+        return lat.matches(DOUBLE_PATTERN)
+                && lon.matches(DOUBLE_PATTERN)
+                && isValidLocation(Double.parseDouble(lat), Double.parseDouble(lon));
+    }
+
+    /**
+     * Checks whether the location lies within the bounds of Singapore.
+     * @param lat Latitude.
+     * @param lon Longitude.
+     * @return Whether the lat and lon fit the requirements.
+     */
+    public static boolean isValidLocation(double lat, double lon) {
+        return isValidLatitude(lat)
+                && isValidLongitude(lon);
+    }
+
+    /**
+     * Checks whether the latitude lies within the bounds of Singapore.
+     * @param lat Latitude.
+     * @return Whether the lat fits the requirements.
+     */
+    public static boolean isValidLatitude(double lat) {
+        return lat >= MIN_LAT && lat <= MAX_LAT;
+    }
+
+    /**
+     * Checks whether the longitude lies within the bounds of Singapore.
+     * @param lon Longitude.
+     * @return Whether the lon fits the requirements.
+     */
+    public static boolean isValidLongitude(double lon) {
+        return lon >= MIN_LON && lon <= MAX_LON;
+    }
+
+    /**
+     * Gets the String value stored within the location.
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Gets the latitude of the location.
+     */
+    public double getLat() {
+        return lat;
+    }
+
+    /**
+     * Gets the longitude of the location.
+     */
+    public double getLon() {
+        return lon;
+    }
```

Comment:
Suggestion to change method name to `getLong()` instead, where long is the short form for longitude.

Reply:
```longitude``` is a better name than the short form.

Reply:
Sorry about the confusion, I meant that `lon` might be confusing and I agree that `longitude` is a better name too.

Reply:
Alright I’ll change it.